### PR TITLE
b2b-smoke-OMS-test

### PR DESCRIPTION
**Pull request by durgesh-verma-spryker**
## PR Description
https://spryker.atlassian.net/browse/CC-21651
OMS

- Process the order up to Closed state
- Process the order, make some products Refunded
- Process the order, make some products returned then refunded

### CODE REVIEW

File: resources/steps/orders_management_steps.robot [hunk 1]
```diff
@@ -151,3 +151,22 @@ Zed: order has the following number of shipments:
     Wait Until Element Is Visible    xpath=//table[@data-qa='order-item-list'][1]
     ${actualShipments}=    Get Element Count    xpath=//table[@data-qa='order-item-list']
     Should Be Equal    '${expectedShipments}'    '${actualShipments}'
+
+Zed: trigger all matching states inside this return:
```
Comment (AntonShopin) [inline resources/steps/orders_management_steps.robot:155]: Do we really need this new keyword? 
Looks like we already have `Zed: trigger all matching states inside xxx order: ` and `Zed: trigger all matching states inside this order:`

File: tests/smoke/smoke_b2b.robot [hunk 2]
```diff
@@ -866,4 +866,61 @@ Refunds
...
+    Yves: 'Thank you' page is displayed
+    Yves: get the last placed order ID by current customer
+    Zed: login on Zed with provided credentials:    ${zed_admin_email}
+    Zed: trigger all matching states inside xxx order:    ${lastPlacedOrder}    Pay
+    Zed: trigger all matching states inside this order:    Skip timeout
+    Zed: trigger all matching states inside this order:    Ship
+    Zed: trigger all matching states inside this order:    Stock update
+    Zed: trigger all matching states inside this order:    Close
+
+Zed_part_OMS_refund_and_return_then_refund
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2b.robot:897]: Please refactor your test. There are 2 ways how to make a refund. The first one is when you create a return -> execute the return and then refund. This case is already covered here https://github.com/spryker/robotframework-suite-tests/blob/master/tests/smoke/smoke_b2b.robot#L564
You need to cover only the second case when we make a refund after the 'delivered' state. 

Also, add in your test assertion, that grand total of the order is reduced on refunded amount. 
Good scenario would be:
* have 2 or more items in the order
* refund only one item from the order -> check that grand total is reduced 
* refund all remaining items using `Zed: trigger all matching states inside this order:` check that grand total is 0 now. 

Please rename your test, don't use 'Zed_part'

File: tests/smoke/smoke_b2b.robot [hunk 3]
```diff
@@ -866,4 +866,61 @@ Refunds
     Zed: trigger all matching states inside this order:    Refund
     Zed: grand total for the order equals:    ${lastPlacedOrder}    €0.00
     [Teardown]    Run keywords    Zed: login on Zed with provided credentials:    ${zed_admin_email}
-    ...    AND    Zed: activate following discounts from Overview page:    20% off storage    10% off minimum order
\ No newline at end of file
+    ...    AND    Zed: activate following discounts from Overview page:    20% off storage    10% off minimum order
+
+Zed_part_OMS_order_close
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2b.robot:871]: 1. Please rename the test. All tests are called like features, don't put `Zed_part_` into the name. A good name would be OMS_close_items
2. Please add [Documetnation] section with the description
3. In your test it would be better to:
* first move to the close state only one of the items -> check that this item is in the closed state and other remain in their states (not closed). With these actions, you'll check that only one item can be 'closed'
* with the next steps move all other items into the closed state using `Zed: trigger all matching states inside this order:`. With these actions, you'll check that all items can be 'closed'
4. At the end check that not the whole order is in the 'closed' state

**Review by AntonShopin — CHANGES_REQUESTED** [review]

File: resources/steps/orders_management_steps.robot (hunk 1 above)
Comment (durgesh-verma-spryker) [reply resources/steps/orders_management_steps.robot:155]: Done as per your suggestions

File: tests/smoke/smoke_b2b.robot (hunk 2 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Done based on your suggestions

File: tests/smoke/smoke_b2b.robot (hunk 3 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Done as per your suggestions, but for point 4- I believe you have added "**not** the whole order" by mistakenly Please confirm if I am wrong.

File: tests/smoke/smoke_b2b.robot [hunk 4]
```diff
@@ -866,4 +866,72 @@ Refunds
...
+    Zed: wait for order item to be in state:    410083    closed
+    Zed: wait for order item to be in state:    421426    closed
+    Zed: go to second navigation item level:    Sales    Orders
+    Zed: perform search by:    ${lastPlacedOrder}
+    Zed: table should contain non-searchable value:    Closed
+
+OMS_Refund_After_Delivery
+    [Documentation]    Process the order delivery state and then initiate the refund
+    Yves: login on Yves with provided credentials:    ${yves_company_user_buyer_email}
+    Yves: create new 'Shopping Cart' with name:    complete+${random}
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2b.robot:909]: please change the name of the new cart. Each time when you create a cart or create a customer or shopping list or ... ->  try to specify unique names (values). In this case, you create a cart with the name `complete+${random}` in your first and second tests. 

When you run tests in a batch,` ${random}` will be the same. If the first test will fail to due different reasons and the cart won't be deleted, in the second test you'll try to create new cart with exactly the same already existing name. It may cause a failure in your tests.

Long story short. In each test add something unique to you values, which is related to your test. In this example cart can be called: `omsRefund+${random}`

File: tests/smoke/smoke_b2b.robot (hunk 3 above)
Comment (AntonShopin) [reply tests/smoke/smoke_b2b.robot]: you are right regarding point #4. I keep this conversation open as one assertion from it is missing (added new comment on it)

File: resources/steps/orders_management_steps.robot (hunk 1 above)
Comment (durgesh-verma-spryker) [reply resources/steps/orders_management_steps.robot]: Removed

File: tests/smoke/smoke_b2b.robot (hunk 4 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot (hunk 3 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: added missing assertion

File: tests/smoke/smoke_b2b.robot [hunk 5]
```diff
@@ -866,4 +866,72 @@ Refunds
...
+    Yves: 'Thank you' page is displayed
+    Yves: get the last placed order ID by current customer
+    Zed: login on Zed with provided credentials:    ${zed_admin_email}
+    Zed: trigger all matching states inside xxx order:    ${lastPlacedOrder}    Pay
+    Zed: trigger all matching states inside this order:    Skip timeout
+    Zed: trigger all matching states inside this order:    Ship
+    Zed: trigger matching state of order item inside xxx shipment:    108278    Stock update
+    Zed: trigger matching state of order item inside xxx shipment:    108278    Close
+    Zed: wait for order item to be in state:    108278    closed
+    Zed: trigger all matching states inside this order:    Stock update
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:901]: Fixed

**Review by AntonShopin — COMMENTED** [review]
The first test "OMS_Close_Items" is ok now, thank you!
The second one is not needed as it fully covered with an already existing one: "Refunds"

You need to extend the second test to cover the following case: 'Process the order, make some products **returned** then refunded'
